src/menu/menu.c

```diff
@@ -893,6 +895,161 @@ menu_hide_submenu(struct server *server, const char *id)
 	}
 }
 
+/*
+ * This is client-send-to-menu
+ * an internal menu similar to root-menu and client-menu
+ *
+ * This will look at workspaces and produce a menu
+ * with the workspace names that can be used with
+ * SendToDesktop, left/right options are included.
+ */
+void
+create_client_send_to_menu(struct server *server)
+{
+	struct menu *menu = menu_get_by_id(server,
+			"client-send-to-menu");
+
+	if (menu) {
+		struct menuitem *item, *next;
+		wl_list_for_each_safe(item, next, &menu->menuitems, link) {
+			item_destroy(item);
+		}
+	} else {
+		menu = menu_create(server, "client-send-to-menu", "Send to...");
+	}
+
+	menu->label = xstrdup("Send to...");
+	menu->size.height = 0;
+
+	if (wl_list_empty(&menu->menuitems)) {
+		struct workspace *workspace;
+
+		wl_list_init(&menu->menuitems);
+		wl_list_for_each(workspace, &server->workspaces, link) {
+			if (!strcmp(workspace->name,
+						server->workspace_current->name)) {
+				current_item = item_create(menu, strdup_printf(">%s<",
+							workspace->name), false);
+			} else {
+				current_item = item_create(menu, workspace->name, false);
+			}
+			fill_item("name.action", "SendToDesktop");
+			fill_item("to.action", workspace->name);
+		}
+
+		current_item = separator_create(menu, "");
+		current_item = item_create(menu, "left", false);
+		fill_item("name.action", "SendToDesktop");
+		fill_item("to.action", "left");
+		current_item = item_create(menu, "right", false);
+		fill_item("name.action", "SendToDesktop");
+		fill_item("to.action", "right");
+	}
+	menu_update_width(menu);
+	wlr_scene_node_set_enabled(&menu->scene_tree->node, false);
+}
+
+/*
+ * This is client-list-combined-menu
+ * an internal menu similar to root-menu and client-menu
+ *
+ * This will look at workspaces and produce a menu
+ * with the workspace name as a separator label
+ * and the titles of the view, if any, below each workspace name.
+ * There is also a window state indicator, similar to osd,
+ * where max, min, and fullscreen are indiciated in one space
+ * always on top/bottom are indicated in one space
+ * and active view is indicated.
+ */
+void
+create_client_list_combined_menu(struct server *server)
+{
+	struct menu *menu = menu_get_by_id(server,
+			"client-list-combined-menu");
+
+	if (menu) {
+		struct menuitem *item, *next;
+		wl_list_for_each_safe(item, next, &menu->menuitems, link) {
+			item_destroy(item);
+		}
+	} else {
+		menu = menu_create(server, "client-list-combined-menu", "Clients...");
+	}
+
+	menu->label = xstrdup("Clients...");
```

Comment:
Same here with leaking memory and I don't think its needed as the label is supplied by the 3rd argument in `menu_create()`.